app/routes/screens/performanceMeter/PerformanceMeter.js

```diff
@@ -1,66 +1,41 @@
-// react
 import React, { Component, PropTypes } from "react";
 
-// react-native
 import { View, Text } from "react-native";
 
-// redux
 import { connect } from "react-redux";
-import { bindActionCreators } from "redux";
 
-// actions
-import { getCurrentAthleteStats } from "../../../store/actions/athlete";
-
-// styles
 import styles from "./styles";
 
 class PerformanceMeter extends Component {
   static propTypes = {
-    currentAthleteStateAction: PropTypes.func.isRequired,
-    currentAthleteID: PropTypes.number.isRequired
+    performance: PropTypes.number.isRequired
   };
 
-  componentWillReceiveProps(nextProps) {
-    if (
-      nextProps.currentAthleteID !== undefined &&
-        nextProps.currentAthleteID !== 0 &&
-        nextProps.currentAthleteID !== this.props.currentAthleteID
-    ) {
-      this.props.currentAthleteStateAction();
-    }
-  }
-
   render() {
     return (
```

Comment:
if(!this.props.performance)
    return <Text>computing</Text>
return ....

and  map null if you are loading